src/backend/api.ml

```diff
@@ -40,3 +40,32 @@ let put ~of_yojson ~to_yojson callback = fun req ->
       | Ok res -> Dream.json ~code:201 (Yojson.Safe.to_string (to_yojson res))
       | Error err -> error err
     )
+
+(* PATCH requests *)
+(* ************************************************************************* *)
+
+let patch ~of_yojson ~to_yojson callback = fun req ->
+  State.get req (fun st ->
+      let%lwt body = Dream.body req in
+      let res =
+        match of_yojson (Yojson.Safe.from_string body) with
+        | exception Yojson.Json_error msg ->
+          Error.(mk @@ invalid_json_body msg)
+        | Error msg ->
+          Error.(mk @@ invalid_json_body msg)
+        | Ok input -> callback req st input
+      in
+      match res with
+      | Ok res -> Dream.json ~code:201 (Yojson.Safe.to_string (to_yojson res))
+      | Error err -> error err
+    )
```

Comment:
This code is identical to the one for `put`, so you could just define it as an alias to it: `let patch = put`.